### PR TITLE
Add ability to blink lighting layer for a specified duration

### PR DESCRIPTION
**Pull request by spidey3**
## Description

This change adds optional functionality to allow blinking an [RGB Lighting Layer](https://docs.qmk.fm/#/feature_rgblight?id=lighting-layers) on for a specified number of milliseconds. The purpose is to make it easy to use a lighting layer to acknowledge events (e.g. a settings change).

To use this new code, it must be enabled using  `#define RGBLIGHT_LAYER_BLINK` e.g. in your config.h.

Kudos to @n8gray for the great RGBLight layers implementation. It is my hope that this small addition will make it even more useful!

## Types of Changes

- [x] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/feature_rgblight.md [hunk 1]
```diff
@@ -228,7 +234,42 @@ bool led_update_user(led_t led_state) {
...
+const rgblight_segment_t PROGMEM _no_layer[] = RGBLIGHT_LAYER_SEGMENTS( {9, 6, HSV_RED} );
+
+const rgblight_segment_t* const PROGMEM _rgb_layers[] =
+    RGBLIGHT_LAYERS_LIST( _yes_layer, _no_layer );
+
+void keyboard_post_init_user(void) {
+    rgblight_layers = _rgb_layers;
+}
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline docs/feature_rgblight.md:255]: This line shouldn't be needed.

Comment (spidey3) [reply docs/feature_rgblight.md]: You are correct; it's needed only in my specific use. I've removed it.

**Comment by drashna** [comment]
Could you pul out the user code, here? 

Otherwise, LGTM.

**Comment by spidey3** [comment]
> Could you pul out the user code, here?

Done.

**Comment by spidey3** [comment]
Gentle ping...

File: docs/feature_rgblight.md [hunk 2]
```diff
@@ -177,6 +177,8 @@ const uint8_t RGBLED_GRADIENT_RANGES[] PROGMEM = {255, 170, 127, 85, 64};
 By including `#define RGBLIGHT_LAYERS` in your `config.h` file you can enable lighting layers. These make
 it easy to use your underglow LEDs as status indicators to show which keyboard layer is currently active, or the state of caps lock, all without disrupting any animations. [Here's a video](https://youtu.be/uLGE1epbmdY) showing an example of what you can do.
 
+### Defining Lighting Layers
```
Comment (noroadsleft) [inline docs/feature_rgblight.md:180]:
```suggestion
### Defining Lighting Layers :id=defining-lighting-layers
```

File: docs/feature_rgblight.md [hunk 3]
```diff
@@ -211,8 +213,12 @@ void keyboard_post_init_user(void) {
     rgblight_layers = my_rgb_layers;
 }
 ```
+Note: For split keyboards with two controllers, both sides need to be flashed when updating the contents of rgblight_layers.
 
-Finally, we enable and disable the lighting layers whenever the state of the keyboard changes:
+### Enabling and disabling lighting layers
```
Comment (noroadsleft) [inline docs/feature_rgblight.md:218]:
```suggestion
### Enabling and disabling lighting layers :id=enabling-lighting-layers
```

File: docs/feature_rgblight.md [hunk 4]
```diff
@@ -228,7 +234,40 @@ bool led_update_user(led_t led_state) {
 }
 ```
 
-Note: For split keyboards with two controllers, both sides need to be flashed when updating the contents of rgblight_layers.
+### Lighting layer blink
```
Comment (noroadsleft) [inline docs/feature_rgblight.md:237]:
```suggestion
### Lighting layer blink :id=lighting-layer-blink
```

**Comment by spidey3** [comment]
> Could you please run `qmk cformat quantum/rgblight.c quantum/rgblight.h` and commit the changes it gives?

Done.

**Pull request merged**